### PR TITLE
Feature/rdm 13146 update document category

### PR DESCRIPTION
**Pull request by sem-hmcts**
### NOTE ###
#1949 **must** be merged before this PR

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-13146
the changes by the following PRs are included here in the PR and thus closed
- https://tools.hmcts.net/jira/browse/RDM-13089
  - #1886 
- https://tools.hmcts.net/jira/browse/RDM-13090
  - #1940
- https://tools.hmcts.net/jira/browse/RDM-13091 
  - #1891
as well as the following draft PRs
- https://tools.hmcts.net/jira/browse/RDM-13124 
  - #[45](https://github.com/hmcts/ccd-test-definitions/pull/45)
  - #1914 
- https://tools.hmcts.net/jira/browse/RDM-13125 
   - #1935 

### Change description ###
Return categories and documents after a document's category has been changed. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sullivangareth — APPROVED** [review]

**Review by CameronGuthrie1 — APPROVED** [review]

File: src/aat/resources/features/F-140 - CategoriesAndDocument endpoint/F-140_CreateCase_hierarchyCollectionComplex_override.td.json [hunk 1]
```diff
@@ -0,0 +1,210 @@
+{
```
[the remark this answers is not in the document]
Comment (marianadpereira) [reply src/aat/resources/features/F-140 - CategoriesAndDocument endpoint/F-140_CreateCase_hierarchyCollectionComplex_override.td.json:1]: Hi Gareth, I tried that, but it doesn't work. Ends up with documents uploaded in the collection and the complex fields, I could not completely override the request body.

File: src/aat/resources/features/F-140 - CategoriesAndDocument endpoint/F-140_CreateCase_hierarchyComplex.td.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
+{
```
[the remark this answers is not in the document]
Comment (marianadpereira) [reply src/aat/resources/features/F-140 - CategoriesAndDocument endpoint/F-140_CreateCase_hierarchyComplex.td.json:1]: Same as above

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -19,7 +19,7 @@ def branchesToSync = ['demo', 'ithc', 'perftest']
 
 // Variables to switch pipeline logic and wiring per type of build
 
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
+def definitionStoreDevelopPr = "PR-1204"  // This doesn't change frequently, but when it does, only change this value.
```
Comment (marianadpereira) [inline Jenkinsfile_CNP:22]: this needs to be  to be reverted before merging

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 4]
```diff
@@ -27,7 +27,7 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1204.service.core-compute-preview.internal/
```
Comment (marianadpereira) [inline charts/ccd-data-store-api/values.preview.template.yaml:30]: this needs to be  to be reverted before merging

**Pull request merged**